ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/AbstractGasLimitSpecification.java

```diff
@@ -40,7 +40,14 @@ public static long deltaBound(final long currentGasLimit) {
     return Long.divideUnsigned(currentGasLimit, GAS_LIMIT_BOUND_DIVISOR);
   }
 
+  /**
+   * Verify that the target gas limit is within the allowed bounds.
+   *
+   * @param targetGasLimit the target gas limit to validate
+   * @return true if within bounds
+   */
   public static boolean isValidTargetGasLimit(final long targetGasLimit) {
-    return DEFAULT_MIN_GAS_LIMIT <= targetGasLimit && DEFAULT_MAX_GAS_LIMIT >= targetGasLimit;
+    // only check the lower bound as the upper bound is Long.MAX_VALUE
+    return targetGasLimit >= DEFAULT_MIN_GAS_LIMIT;
```

Comment:
Can you point to which tests this is covered by please?

Reply:
ethereum/core/src/test/java/org/hyperledger/besu/ethereum/mainnet/TargetingGasLimitCalculatorTest.java

Reply:
Thanks. I find this errorprone check odd - wouldn't this code break if we error changed the DEFAULT_MAX to be something other than Long.MAX_VALUE?

Reply:
agree it seems safer to suppress this warning

Reply:
To counter-argue, if we jump from long to BigInteger for example, then the targetGasLimit would also change from long to BigInteger. What is the value `DEFAULT_GAS_MAX_LIMIT` as per the spec? Here it seems to be bounded by the variable type we are using i.e. Long.MAX_VALUE.

I'll close this PR and will add the suppression when we bring in the updated errorprone library dependency.

Reply:
Can't find a max value in the spec, was looking here: https://ethereum.github.io/execution-specs/src/ethereum/shanghai/fork.py.html#ethereum.shanghai.fork.check_gas_limit:0

It's UInt so specified as arbitrary size https://ethereum.github.io/execution-specs/src/ethereum/base_types.py.html

>  if we ever changed the DEFAULT_MAX to be something other than Long.MAX_VALUE

I was more thinking if a future spec introduced a specific max value to enforce something in the protocol.

Reply:
Agreed. I'll add a unit test instead to cover the max value.